src/NuGet.Core/NuGet.Protocol.Core.Types/Resources/IPackageSearchResource.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Protocol.Core.Types
+{
+    public interface IPackageSearchResource : INuGetResource
```

Comment:
It's much cleaner concept for contracts with method signature. It doesn't imply semi-implemented object behind. So that one can implement multiple resources with one class for instance.

Perhaps I missed something in resource design. Any particular reason why the rest of resources are abstract?